### ISSUE TITLE
Document.FromJson doesn't support unsigned longs

### ISSUE THREAD

**Issue by flowerbits** [aws/aws-sdk-net#387]
Primitives and DynamoDB easily support large numbers, however the third party json reader doesn't support unsigned numbers.

aws-sdk-net/docgenerator/SDKDocGeneratorLib/ThirdParty/Json/JsonReader.cs,

.NET will throw an overflow exception when trying to do Int64.parse on this example ulong:
16849433912163548285

As a result, the above number won't be included properly in a document retrieved by Document.FromJson

**Comment by PavelSafronov** [comment]
In the below code, the value is parsed incorrectly as 0, though I am not encountering an overflow exception.
```
ulong original = 16849433912163548285;
var doc = Document.FromJson(@"{""ULong"": " + original  + " }");
var parsed = doc["ULong"].AsULong();
Assert.AreEqual(original, parsed);
```

It appears that the JSON reader we are using doesn't support ulong at all.

**Comment by flowerbits** [comment]
That's correct. The library doesn't use Int64.Parse. 

Also, the updated library has support for uint64, so would it be possible to simply update the library to the latest version?

I think it's called litjson, and is found here:
https://github.com/lbv/litjson/

**Comment by flowerbits** [comment]
That's correct. The library doesn't use Int64.Parse. 

Also, the updated library has support for uint64, so would it be possible to simply update the library to the latest version?

I think it's called litjson, and is found here:
https://github.com/lbv/litjson/

**Comment by sstevenkang** [comment]
This has been fixed in commit [aef2e1705](https://github.com/aws/aws-sdk-net/commit/aef2e1705972cc896dfe54f677b57fe2123817c9).
